homeassistant/components/qbittorrent/__init__.py

```diff
@@ -1 +1,85 @@
 """The qbittorrent component."""
+import asyncio
+import logging
+
+from qbittorrent.client import LoginRequired
+from requests.exceptions import RequestException
+
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
+from homeassistant.const import CONF_PASSWORD, CONF_URL, CONF_USERNAME
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import config_per_platform
+
+from .client import create_client
+from .const import DATA_KEY_CLIENT, DATA_KEY_NAME, DOMAIN
+
+PLATFORMS = ["sensor"]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the Qbittorrent component."""
+    hass.data.setdefault(DOMAIN, {})
+
+    # Import configuration from sensor platform
+    config_platform = config_per_platform(config, "sensor")
+    for p_type, p_config in config_platform:
+        if p_type != DOMAIN:
+            continue
+
+        hass.async_create_task(
+            hass.config_entries.flow.async_init(
+                DOMAIN,
+                context={"source": SOURCE_IMPORT},
+                data=p_config,
+            )
+        )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up Qbittorrent from a config entry."""
+    name = "Qbittorrent"
+    try:
+        client = await hass.async_add_executor_job(
+            create_client,
+            entry.data[CONF_URL],
+            entry.data[CONF_USERNAME],
+            entry.data[CONF_PASSWORD],
+        )
+    except LoginRequired:
+        _LOGGER.error("Invalid authentication")
+        return False
+
+    except RequestException as err:
+        _LOGGER.error("Connection failed")
+        raise ConfigEntryNotReady from err
+
+    hass.data[DOMAIN][entry.data[CONF_URL]] = {
```

Comment:
Use the entry_id as the index

```suggestion
    hass.data[DOMAIN][entry.entry_id] = {
```